src/resolve.rs

```diff
@@ -19,39 +19,42 @@ pub(crate) trait ResolveOrDefault<TContext, TOutput> {
 /// Will return a `None` if it unable to resolve.
 pub(crate) trait MaybeResolve<T> {
     /// Resolve a dimension that might be dependent on a context, with `None` as fallback value
-    fn maybe_resolve(self, context: T) -> T;
+    fn maybe_resolve(&self, context: T) -> T;
 }
 
 impl MaybeResolve<Option<f32>> for Dimension {
     /// Converts the given [`Dimension`] into a concrete value of points
     ///
     /// Can return `None`
-    fn maybe_resolve(self, context: Option<f32>) -> Option<f32> {
+    fn maybe_resolve(&self, context: Option<f32>) -> Option<f32> {
         match self {
-            Dimension::Points(points) => Some(points),
+            Dimension::Points(points) => Some(*points),
             // parent_dim * percent
             Dimension::Percent(percent) => context.map(|dim| dim * percent),
-            _ => None,
+            Dimension::Auto | Dimension::Undefined => None,
+
+            #[cfg(feature = "std")]
+            Dimension::Calc(calc) => calc.maybe_resolve(context),
```

Comment:
Does this mean that Calc only work with the `std` feature enabled?

Reply:
Currently yes, because it requires `Box`. Maybe `alloc` would be enough though.